multiview/embed/mvmds.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from multiview.embed.base import BaseEmbed
+from multiview.utils.utils import check_Xs
+import warnings
+import numpy as np
+from sklearn.metrics import euclidean_distances
+
+
+class MVMDS(BaseEmbed):
+
+    """"
+    Classical Multiview Multidimensional Scaling for jointly reducing
+    the dimensions of multiple views of data. A euclidean distance matrix
+    is created for each view, double centered, and the k largest common
+    eigenvectors are returned based on the algorithm proposed by the
+    following paper:
+
+    https://www.sciencedirect.com/science/article/pii/S016794731000112X
+
+    """
+
+    def __init__(self, n_components=None):
+
+        super().__init__()
+        self.components = None
+        self.n_components = n_components
+
+    def _cpc(self, x):
+
+        """
+        Finds Stepwise Estimation of Common Principal Components as described
+        by common Trendafilov implementations based on the following paper:
+
+        https://www.sciencedirect.com/science/article/pii/S016794731000112X
+
+        Parameters
+        ----------
+
+        x: List of matrices, each with number of rows, n
+
+        it: Number of common principal component stepwise iterations
+
+        Returns
+        -------
+
+        Components: Desired Common Principal Components
+
+        """
+        n = p = x.shape[1]
+
+        views = len(x)
+
+        n_num = np.array([n] * views)/np.sum(np.array([n] * views))
+
+        Components = np.zeros((p, self.n_components))
+
+        pi = np.eye(p)
+
+        s = np.zeros((p, p))
+
+        it = 15
+
+        for i in np.arange(views):
+            s = s + (n_num[i] * x[i])
+
+        e1, e2 = np.linalg.eigh(s)
+
+        q0 = e2[:, ::-1]
+
+        for i in np.arange(self.n_components):
```

Comment:
add in line comments throughout this function. there's a lot happening and it's hard to read/understand what the code is doing without them

Reply:
or better yet maybe break this up into smaller functions if possible?

Reply:
with descriptive names

Reply:
Will keep splitting it up as an issue. I wrote this function straight from the pseudocode trendafilov gave in his paper. Can think about how to go about splitting it up